File: backend/Persistence/Models/PaymentInformation.cs
using System;
using System.Collections.Generic;

namespace Persistence.Models;

public partial class PaymentInformation
{
    public Guid Id { get; set; }

    public int Currency { get; set; }

    public int PaymentFrequency { get; set; }

    public double Cost { get; set; }

    public bool Negotiable { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public virtual ICollection<Property> Properties { get; set; } = new List<Property>();
}

File: mobile/lib/src/features/dashboard/data/models/payment_information.dart
import 'package:mobile/src/features/dashboard/domain/domain.dart';

class PaymentInformationModel extends PaymentInformation {
  const PaymentInformationModel(
      {required super.id,
      required super.createdAt,
      required super.updatedAt,
      required super.currency,
      required super.paymentFrequency,
      required super.cost,
      required super.negotiable});

  factory PaymentInformationModel.fromJson(Map<String, dynamic> json) {
    return PaymentInformationModel(
      id: json['id'],
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
      currency: json['currency'],
      paymentFrequency: json['paymentFrequency'],
      cost: json['cost'],
      negotiable: json['negotiable'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt.toIso8601String(),
      'currency': currency,
      'paymentFrequency': paymentFrequency,
      'cost': cost,
      'negotiable': negotiable,
    };
  }
}

